packages/caravan-fees/src/cpfp.ts

```diff
@@ -0,0 +1,105 @@
+import { Transaction } from "bitcoinjs-lib-v5";
+import { unsignedMultisigTransaction } from "@caravan/bitcoin";
+import BigNumber from "bignumber.js";
+import { CPFPOptions, UTXO } from "./types";
+import { estimateVirtualSize } from "./utils";
+
+/**
+ * Prepare a CPFP (Child-Pays-for-Parent) transaction.
+ * This function doesn't broadcast the transaction, it just prepares it.
+ * @param options CPFP options including the parent transaction and new fee rate
+ * @returns A new transaction ready for signing and broadcasting
+ */
+export function prepareCPFPTransaction(options: CPFPOptions): Transaction {
+  const {
+    parentTransaction,
+    newFeeRate,
+    availableUTXOs,
+    destinationAddress,
+    network,
+    multisigDetails,
+  } = options;
+
+  // Find suitable output from parent transaction to spend
+  const parentOutput = findSuitableParentOutput(
+    parentTransaction,
+    availableUTXOs
+  );
+  if (!parentOutput) {
+    throw new Error("No suitable output found in parent transaction for CPFP");
+  }
+
+  const childSize = estimateChildTransactionSize({
+    inputCount: 1, // We're spending one output from the parent
+    outputCount: 2, // Destination output and potentially change
+    addressType: multisigDetails.addressType,
+    requiredSigners: multisigDetails.requiredSigners,
+    totalSigners: multisigDetails.totalSigners,
+  });
+  // Calculate required fee for both transactions
+  const combinedSize = parentTransaction.virtualSize() + childSize;
+  const requiredFee = new BigNumber(combinedSize).multipliedBy(
+    newFeeRate.satoshisPerByte
+  );
+
+  // Calculate amount to send to destination
+  const outputAmount = new BigNumber(parentOutput.value).minus(requiredFee);
+
+  if (outputAmount.isLessThan(546)) {
+    // Check if output would be dust
+    throw new Error("CPFP transaction would create a dust output");
+  }
+
+  const inputs = [{ ...parentOutput, txid: parentTransaction.getId() }];
+  const outputs = [{ address: destinationAddress, amountSats: outputAmount }];
+
+  // Use Caravan's unsignedMultisigTransaction function to create the new transaction
+  return unsignedMultisigTransaction(network, inputs, outputs, true); // true to enable RBF
+}
+
+/**
+ * Find a suitable output from the parent transaction to use for CPFP.
```

Comment:
Might be worthwhile to add what it means to be a "suitable output". I.e. how does this function find a suitable parent output?

Reply:
So as discussed we are now aspect the consumer code to handle and provide us with the suitable output ... [here](https://github.com/caravan-bitcoin/caravan/commit/77f0cbeb67a274b0aea5a69e87168b9ac1e2304a#diff-852fd83c241ed78594dc29389ec04d1425d784e665c706434a8de046f7daa85dR47)